lldb/source/Plugins/TypeSystem/Clang/TypeSystemClang.cpp

```diff
@@ -3742,7 +3753,9 @@ bool TypeSystemClang::IsPossibleDynamicType(lldb::opaque_compiler_type_t type,
               if (metadata)
                 success = metadata->GetIsDynamicCXXType();
               else {
-                is_complete = GetType(pointee_qual_type).GetCompleteType();
+                // Make sure completion has actually completed the type.
+                is_complete = GetType(pointee_qual_type).GetCompleteType() &&
+                              HasCompleteDefinition(cxx_record_decl);
```

Comment:
Which action could have changed the result of `HasCompleteDefinition(cxx_record_decl)` since line 3747?

Reply:
In the call to `GetCompleteType` (which does much more than the name suggests)